yt_dlp/extractor/theplatform.py

```diff
@@ -297,6 +300,27 @@ def _real_extract(self, url):
 
         formats, subtitles = self._extract_theplatform_smil(smil_url, video_id)
 
+        note = 'Downloading m3u8 information'
+        errnote = 'Failed to download m3u8 information'
+        # on at least one site reading just the SMIL data misses some formats as well as subtitles.
+        # Therefore also read the m3u8 file and add those formats and subtitles
+        m3u8_url = url.split("&")[0]
+        res = self._download_webpage_handle(
+            m3u8_url, video_id,
+            note=note,
+            errnote=errnote,
+            fatal=False)
+        if res is not False:
+            m3u8_doc, urlh = res
+            m3u8_url = urlh.geturl()
+            if m3u8_url.split("?")[0][-4:].lower() == "m3u8":
```

Comment:
use `determine_ext` for something like this

and use single quotes for all strings

Reply:
Noted, will fix